src/arrays.jl

```diff
@@ -320,6 +320,46 @@ end
 Base.fill(x::UnionAbstractQuantity, dims::Dims...) = QuantityArray(fill(ustrip(x), dims...), dimension(x), typeof(x))
 Base.fill(x::UnionAbstractQuantity, t::Tuple{}) = QuantityArray(fill(ustrip(x), t), dimension(x), typeof(x))
 
+# Will be overloaded by `DynamicQuantitiesLinearAlgebraExt`:
+_norm(_) = error("Please load the `LinearAlgebra.jl` package.")
+
+# Define isapprox for vectors of Quantity's
+function Base.isapprox(
+    u::AbstractArray{Q1},
+    v::AbstractArray{Q2};
+    atol=new_quantity(Q1, zero(T1), dimension(first(u))),
+    rtol=Base.rtoldefault(T1),
+    nans::Bool=false,
+    norm::F=_norm
+) where {T1,D1,Q1<:UnionAbstractQuantity{T1,D1},T2,D2,Q2<:UnionAbstractQuantity{T2,D2},F<:Function}
+    d = norm(u - v)
+    if isfinite(d)
+        return d <= max(atol, rtol*max(norm(u), norm(v)))
+    else
+        # Fall back to a component-wise approximate comparison
+        return all(ab -> isapprox(uv[1], uv[2]; rtol=rtol, atol=atol, nans=nans), zip(u, v))
```

Comment:
The `zip` is dangerous because if you zip two arrays which have different lengths, it won't throw an error, it will just go until one of the arrays runs out of elements. I think `eachindex(u, v)` is safer.

Reply:
Pushed code to fix this.

Reply:
This pattern is pretty much directly ripped from the method in LinearAlgebra.jl. This case is something I'd thought about, but you only reach this line after completing the previous `d = norm(u - v)` calculation, where non-equal lengths would definitely be caught.

I can take a look at making this change. If we have a better way, I think I'd also recommend we make the same argument upstream in a PR to LinearAlgebra.jl.